### ISSUE TITLE
`Intl.supportedValuesOf('timeZone')` for UTC, Etc/GMT*, and other non-contintent/non-ocean time zone identifiers

### ISSUE THREAD

**Issue by justingrant** [tc39/ecma402#778]
As part of work on [proposal-canonical-tz](https://github.com/tc39/proposal-canonical-tz), I noticed that JS engines have very different results from `Intl.supportedValuesOf('timeZone')` for time zones that are not associated with a continent or ocean. 

V8 does not return any of these zones. Safari returns only one: UTC. Firefox returns 40 zones, including "UTC", Etc/ zones like "Etc/GMT+11", and a few others like "EST" and "CET".

Should the 402 spec be more prescriptive about what kinds of canonical time zone identifiers should vs. should not be returned by this API?

```
Intl.supportedValuesOf('timeZone').filter(z => !['Asia', 'Africa', 'Europe', 'Australia', 'America', 'Antarctica', 'Atlantic', 'Pacific', 'Indian', 'Arctic'].some(c => z.startsWith(c)))

// => Chrome: []
// => Safari: ["UTC"]
// => Firefox: [
  "CET",
  "CST6CDT",
  "EET",
  "EST",
  "EST5EDT",
  "Etc/GMT+1",
  "Etc/GMT+10",
  "Etc/GMT+11",
  "Etc/GMT+12",
  "Etc/GMT+2",
  "Etc/GMT+3",
  "Etc/GMT+4",
  "Etc/GMT+5",
  "Etc/GMT+6",
  "Etc/GMT+7",
  "Etc/GMT+8",
  "Etc/GMT+9",
  "Etc/GMT-1",
  "Etc/GMT-10",
  "Etc/GMT-11",
  "Etc/GMT-12",
  "Etc/GMT-13",
  "Etc/GMT-14",
  "Etc/GMT-2",
  "Etc/GMT-3",
  "Etc/GMT-4",
  "Etc/GMT-5",
  "Etc/GMT-6",
  "Etc/GMT-7",
  "Etc/GMT-8",
  "Etc/GMT-9",
  "Factory",
  "HST",
  "MET",
  "MST",
  "MST7MDT",
  "PST8PDT",
  "UTC",
  "WET"
]
```

**Comment by sffc** [comment]
TG2 discussion: https://github.com/tc39/ecma402/blob/master/meetings/notes-2023-05-04.md#intlsupportedvaluesoftimezone-for-utc-etcgmt-and-other-non-contintentnon-ocean-time-zone-identifiers-778

**Comment by anba** [comment]
[Intl.supportedValuesOf](https://tc39.es/ecma402/#sec-intl.supportedvaluesof) calls [AvailableCanonicalTimeZones](https://tc39.es/ecma402/#sec-availablecanonicaltimezones), which computes the result based on all supported time zone and link names. 

- `CET`, `EET`, `MET`, and `WET` are defined time zone zones in the [europe file](https://github.com/eggert/tz/blob/5a220576fde1f7d3b6f716702d93fae08dbd2268/europe#L736-L739).
- `CST6CDT`, `EST`, `EST5EDT`, `HST`, `MST`, `MST7MDT`, and `PST8PDT` are defined as time zone zones in the [northamerica file](https://github.com/eggert/tz/blob/5a220576fde1f7d3b6f716702d93fae08dbd2268/northamerica#L200-L206).
- `Etc/GMT-14` to `Etc/GMT+12` are defined as time zone zones in the [etcetera file](https://github.com/eggert/tz/blob/5a220576fde1f7d3b6f716702d93fae08dbd2268/etcetera#L54-L79).
- `Etc/UTC` and `Etc/GMT` are also defined in the [etcetera file](https://github.com/eggert/tz/blob/5a220576fde1f7d3b6f716702d93fae08dbd2268/etcetera#L21-L34). Both are canonicalised to `UTC` in [CanonicalizeTimeZoneName](https://tc39.es/ecma402/#sec-canonicalizetimezonename). 
- `Factory` is from the [factory file](https://github.com/eggert/tz/blob/5a220576fde1f7d3b6f716702d93fae08dbd2268/factory#L12). IIRC it's just a placeholder time zone, but because it's supported in all browsers (`new Intl.DateTimeFormat("en", {timeZone: "Factory"}).resolvedOptions().timeZone`), I've never [removed the support](https://searchfox.org/mozilla-central/rev/e77d89c414eac63a295a2124600045775a6f4715/js/src/builtin/intl/make_intl_data.py#3034-3035) for it. 

So all additional time zones returned by Firefox are correct and should also be returned by the other browsers, as long as the time zone id is supported by `Intl.DateTimeFormat`.

**Comment by sffc** [comment]
In the latest TZDB, thanks to @justingrant, the legacy zones like "WET" are now links in `backward`.

I think that should make this issue easier to resolve, now? `Etc/*` zones should be included in `supportedValuesOf`.

**Comment by justingrant** [comment]
> I think we should go ahead and make a PR.

AFAIK, no PR for the ECMA 402 spec is needed. But JSC and V8 don't currently follow the current spec because they omit `Etc/*` (both JSC and V8) zones and omit (V8 only) `UTC`.

So changing this is really a matter of convincing implementers to follow the spec.

**Comment by justingrant** [comment]
On the other hand, if we want to reify the no-Etc behavior of V8/JSC into the spec, that would require a normative PR.

Should we discuss this at the next TG2 meeting?

**Comment by sffc** [comment]
I think impls should return Etc/* time zones in supportedValuesOf, and not WET/CST/...

Is this what the spec currently requires? If so, then maybe we only need to add some tests.

**Comment by justingrant** [comment]
> Is this what the spec currently requires? 

Yes. There is no filtering out of `Etc/*` and `UTC` in the [spec](https://tc39.es/ecma402/#sec-availableprimarytimezoneidentifiers). All primary time zone identifiers are supposed to be returned, but today only SpiderMonkey is following the spec.
